### ISSUE TITLE
nrepl 1.3.0 timeout when used via maven

### ISSUE THREAD

**Issue by bcorrigan** [nrepl/nrepl#358]
Hello, when upgrading to nrepl 1.3.0 from 1.2.0 I noticed an issue, which is that it times out after around 60 seconds. This is absent in 1.2.0.

## Expected behavior

"mvn clojure:nrepl" should start an nrepl indefinitely without timeout.

## Actual behavior

"mvn clojure:nrepl" times out after 60 seconds.

## Steps to reproduce the problem

I've attached a pom.xml. To reproduce:

- Install maven
- Copy the pom.xml to an empty directory
- Execute "mvn clojure:nrepl" in the empty directory.
- Wait for 60 seconds

## Environment & Version information

java -version:

openjdk version "22.0.2" 2024-07-16

mvn --version: 

openjdk version "22.0.2" 2024-07-16

uname -a:

Linux t16 6.10.5-native_intel-xanmod1-1 #1 SMP PREEMPT_DYNAMIC Thu, 15 Aug 2024 08:52:54 +0000 x86_64 GNU/Linux

Everything else as per the pom deps really. 

[pom.zip](https://github.com/user-attachments/files/16661778/pom.zip)

**Comment by alexander-yakushev** [comment]
Hah, apparently this is a side effect of #345. By addressing #219 we actually broke cases where it was desirable for Future threads to be keeping the process alive.

**Comment by bcorrigan** [comment]
Thanks for the response - sure, I will create an issue later today on the maven plugin repo. I think you are right it is not updated often, but it is worth a go!

**Comment by alexander-yakushev** [comment]
I managed to Lisp-inject the fix into your `pom.xml`. This is probably not something you want to be doing, but still, it's fun 😄 .
```xml
<middleware>cider.nrepl/wrap-xref)))) @(promise) '((((</middleware>
```

**Comment by bcorrigan** [comment]
Wow nice, that's indeed quite evil! But also means that the "fix" is hopefully quite simple for the clojure maven folks.

**Comment by alexander-yakushev** [comment]
Actually, there is a knob you can use. It's still a bit hackish. You create a file named `never-stop.clj` in the parent directory of the project with this content:
```clj
(future @(promise))
```
And then add this to pom.xml somewhere inside the plugin `<configuration>` tag:
```xml
<replScript>never-stop.clj</replScript>
```
Still not ideal to do the extra steps but can cover you meanwhile. That, or just downgrading to 1.2.0, obviously.

**Comment by bcorrigan** [comment]
Brilliant, thanks very much for the help! As it is maven, these steps only make it about 1% more ugly anyway heh.

**Comment by talios** [comment]
I've just released 1.9.3 of `clojure-maven-plugin` which now includes support for this via a new `<keepRunning>true</keepRunning>` property on the `nrepl` goal.

**Comment by alexander-yakushev** [comment]
Awesome, thank you for addressing this!
